RHDEVDOCS-6339: Content creation for CMP (#88418)
openshift/openshift-docs: modules/con_enabling-config_management_plugins_in_an_argo_cd_cr.adoc

// Module included in the following assemblies:
//
// * argocd_instance/argo-cd-cr-component-properties.adoc

:_mod-docs-content-type: CONCEPT

> [id="enabling-config_management_plugins_argocd_{context}"]

Reviewer: ```suggestion
[id="enabling-config-management-plugins-argocd_{context}"]
```

Author: Fixed.

> [id="enabling-config_management_plugins_in_an_argo_cd_cr_{context}"]

Reviewer 2: ```suggestion
[id="enabling-config_management_plugins_argocd_{context}"]
```

I'd suggest to mkae this id shorter

Author: Fixed.

= Enabling Config Management Plugins in an Argo CD CR

Argo CD provides support for Helm, Jsonnet, and Kustomize as built-in config management tools. To use a different config management tool, or to enable features not provided by the built-in config management tools, you can use the Config Management Plugin (CMP).

> In Argo CD, the Config Management Plugin (CMP) is specified as a sidecar container for the Argo CD repo server container. For more information, see _Config Management Plugins_.

Reviewer 2: ```suggestion
In Argo CD, the CMP is specified as a sidecar container for the Argo CD repo server container. For more information, see "Config Management Plugins".
```

- Acronym is already expanded above.
- See - https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#lead-in-link-sentence-style-for-non-xref-situations

Author: Fixed.

> In the {gitops-title} Operator, the Config Management Plugin (CMP) is specified as a sidecar container for the Argo CD repo server container. For more information, see Config Management Plugins.

Reviewer 3: This looks a little redundant after reading the following passage

Author: Are you referring to the entire passage being redundant and want me to remove it from the content @varshab1210?

Reviewer 4: Hmm, the last `Config Management Plugins` in that sentence should be a link to the upstream docs: https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/

Author: Fixed.

> In the {gitops-title} Operator, you can configure the Config Management plugin as a sidecar container in the Argo CD custom resource (CR). When configuring the sidecar container, you can specify an off-the-shelf or custom-built container image.

Reviewer 4: This should include the info that you can omit explicitly specifying the image.

```
When configuring the sidecar container, you can specify an off-the-shelf or custom-built container image, or you can omit specifying the image in which case the image used for the repo server will also be used for the plugin
```

Author: Fixed.

To configure a sidecar container in the {gitops-title} Operator, add the `.spec.repo.sidecarContainers` key in the Argo CD CR. 

> .Example for `Config Management Plugin`

Reviewer 2: ```suggestion
.Example Config Management Plugin configuration
```

Author: Fixed.

[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:

>  name: my-plugin # <1>

Reviewer 4: I'd change this to something like `name: my-argocd` since in general there's more to the argocd cr than just the plugin.

Author: Thanks for the feedback @jopit, I have changed the name of the Argo CD instance. Fixed.

spec:
 repo:
   sidecarContainers:
     - name: my-cmp # <2>
       command: [/var/run/argocd/argocd-cmp-server]
       image: my-image
       securityContext:
         runAsNonRoot: true
         runAsUser: 999
       volumeMounts: # <3>
         - mountPath: /var/run/argocd
           name: var-files
         - mountPath: /home/argocd/cmp-server/plugins
           name: plugins
         - mountPath: /tmp
           name: tmp
         - mountPath: /home/argocd/cmp-server/config/plugin.yaml
           subPath: plugin.yaml
           name: cmp-plugin
----
<1> Specify the name of a Config Management plugin CR defined in an Argo CD CR.
<2> Specify the name of a sidecar container used in the repo server.

> <3> Specify the name of a volume mount used in the repo server.

Reviewer 4: @jgwest yes you are correct, this is part of the ArgoCD CR. I missed that in my review.

> <3> Specify the name of a volume mount used in the repo server.

Author: Fixed.

> <3> Specifies the name of volume mounts used in the repo server.

Reviewer: We should be parallel here with the others, so use either "Specify", or change the name of the others to "Specifies. . ."

```suggestion
<3> Specifies the name of volume mounts used in the repo server.
```

Author: Keeping it as *Specify* to ensure consistency.

> <3> Specify the name of volume mounts used in the repo server.

Reviewer 2: ```suggestion
<3> Specifies the name of volume mounts used in the repo server.
```

because it's not something user-replaceable

Author: Fixed.

> <3> Specify the name of a volume mount used in the repo server.

Reviewer 3: ```suggestion
<3> Specify the name of volume mounts used in the repo server.
```

Author: Fixed.
